source/dem/dem.cc

```diff
@@ -272,24 +284,112 @@ DEMSolver<dim>::cell_weight(
   return 0;
 }
 
+template <int dim>
+unsigned int
+DEMSolver<dim>::cell_weight_with_mobility_status(
+  const typename parallel::distributed::Triangulation<dim>::cell_iterator &cell,
+  const typename parallel::distributed::Triangulation<dim>::CellStatus status,
+  const unsigned int &mobility_status) const
+{
+  // Assign no weight to cells we do not own.
+  if (!cell->is_locally_owned())
+    return 0;
+
+  const unsigned int particle_weight =
+    parameters.model_parameters.load_balance_particle_weight;
+
+  // Applied a factor on the particle weight regards the mobility status
+  // Factor of 1 when mobile cell
+  double alpha = 1.0;
+  if (mobility_status == disable_contact_object.active)
+    {
+      alpha = parameters.model_parameters.active_load_balancing_factor;
+    }
+  else if (mobility_status == disable_contact_object.inactive)
+    {
+      alpha = parameters.model_parameters.inactive_load_balancing_factor;
+    }
+
+  switch (status)
+    {
+      case parallel::distributed::Triangulation<dim>::CELL_PERSIST:
```

Comment:
As done in the "original" function, this is because CELL_PERSIST and CELL_REFINE will both execute the line 318 to 320. A case not having a break will execute until there's a break.
https://stackoverflow.com/questions/33222437/switch-case-statement-without-break

Reply:
Hey, you did that part :p Thanks to git blame

Reply:
Oh it sounds kinda passive-aggresive, not my intention hahahaha Sorry 🫠